tests/nix-shell.sh

```diff
@@ -58,6 +58,11 @@ output=$(NIX_PATH=nixpkgs=shell.nix nix-shell --pure -p foo bar --run 'echo "$(f
 output=$(NIX_PATH=nixpkgs=shell.nix nix-shell --pure -p foo --argstr fooContents baz --run 'echo "$(foo)"')
 [ "$output" = "baz" ]
 
+# Test nix-shell -p with custom nixpkgs expression that does not expect
+# the (new) isNixShell attribute
```

Comment:
```suggestion
# Test nix-shell -p with older versions of Nixpkgs that do not expect
# the (new) isNixShell attribute.
```

We don't support `nixpkgs` pointing to things that are not Nixpkgs. We do want to support older versions of Nixpkgs however.